### ISSUE TITLE
Calico Windows - after a while Windows node will lose access to the internal network

### ISSUE THREAD

**Issue by MyKEms** [projectcalico/calico#4664]
We have windows worker nodes joined to k8s cluster. When it's deployed it works as expected. However by a time (1-4 weeks) windows decide to not communicate in LAN anymore but k8s still works, however it depends still on LAN which is inaccessible and by this will stop work DNS resolution on windows node into LAN + all pods on affected windows node can't use `core-dns` resolution and ping any resource.

I did some inspection and tried to simulate this issue and it led me when `kubelet service` is started the LAN will become after some seconds inaccessible. I realized it's caused by wrong routing which calico will probably reconfigure by mistake.

## Expected Behavior

Calico should not reconfigure routing table for interface `vEthernet (Calico_ep)` with the same metric as the main `vEthernet (Ethernet0)` which is connected to LAN.

## Current Behavior

Randomly, by a time calico will reconfigure `vEthernet (Calico_ep)` with same metric like main `vEthernet (Ethernet0)` which is connected to LAN. 

## Possible Solution

1. List the interfaces and check if `vEthernet (Calico_ep)` has same metric like `vEthernet (Ethernet0)`

```pwsh
Get-NetIPInterface

ifIndex InterfaceAlias                  AddressFamily NlMtu(Bytes) InterfaceMetric Dhcp     ConnectionState PolicyStore
------- --------------                  ------------- ------------ --------------- ----     --------------- -----------
30      vEthernet (Calico_ep)           IPv6                  1500              15 Enabled  Connected       ActiveStore
16      vEthernet (Ethernet0)           IPv6                  1500              15 Enabled  Connected       ActiveStore
21      vEthernet (nat)                 IPv6                  1500            5000 Enabled  Connected       ActiveStore
1       Loopback Pseudo-Interface 1     IPv6            4294967295              75 Disabled Connected       ActiveStore
30      vEthernet (Calico_ep)           IPv4                  1450              15 Disabled Connected       ActiveStore
16      vEthernet (Ethernet0)           IPv4                  1450              15 Disabled Connected       ActiveStore
21      vEthernet (nat)                 IPv4                  1450            5000 Disabled Connected       ActiveStore
1       Loopback Pseudo-Interface 1     IPv4            4294967295              75 Disabled Connected       ActiveStore
```

2.  Set for `vEthernet (Calico_ep)` lower priority so windows worker node can again communicate in LAN network first, then calico.

```pwhs
Set-NetIPInterface -InterfaceIndex 30 -InterfaceMetric 20
```

3. Result should be lower priority for `vEthernet (Calico_ep)`:

```pwsh
ifIndex InterfaceAlias                  AddressFamily NlMtu(Bytes) InterfaceMetric Dhcp     ConnectionState PolicyStore
------- --------------                  ------------- ------------ --------------- ----     --------------- -----------
30      vEthernet (Calico_ep)           IPv6                  1500              20 Enabled  Connected       ActiveStore
16      vEthernet (Ethernet0)           IPv6                  1500              15 Enabled  Connected       ActiveStore
21      vEthernet (nat)                 IPv6                  1500            5000 Enabled  Connected       ActiveStore
1       Loopback Pseudo-Interface 1     IPv6            4294967295              75 Disabled Connected       ActiveStore
30      vEthernet (Calico_ep)           IPv4                  1450              20 Disabled Connected       ActiveStore
16      vEthernet (Ethernet0)           IPv4                  1450              15 Disabled Connected       ActiveStore
21      vEthernet (nat)                 IPv4                  1450            5000 Disabled Connected       ActiveStore
1       Loopback Pseudo-Interface 1     IPv4            4294967295              75 Disabled Connected       ActiveStore
```

4. Once configured, windows worker node can communicate over LAN, DNS resolution works as expected in Windows Worker node and internal network is accessible together with calico k8s network.

## Steps to Reproduce (for bugs)

I did not find pattern when it happens. I did not touched any system related stuff. So right now I don't know how to reproduce this.

## Context

I don't think Calico interface should have same metric like LAN interface otherwise it causes issues to communicate over LAN for Windows node when this issue happend.

## Your Environment

* Calico version: Windows Version 3.19.1 Cluster Version: v3.19.0, Cluster Type: k8s,bgp,kubeadm,kdd,win
* Orchestrator version (e.g. kubernetes, mesos, rkt): k8s
* Operating System and version: Microsoft Windows Server 2019 Standard, 10.0.17763 Build 17763

**Comment by song-jiang** [comment]
Thanks @MyKEms for the details. Very interesting issue. 

IUUC `vEthernet (Calico_ep)` is derived from and connected to host eth interface within the vSwitch. 
- I don't understand how interfaceMetric value affects the outcome of the connection. Especially it was working for a few weeks. Anything has changes since last time it worked? 

- Even we pin down that interfaceMetric is the issue, there is no API exposed by HNS that Calico can use to program the value.

**Comment by MyKEms** [comment]
Actually there is no change at all. Just deployments rotating pods = spawn and kill (used for building).

We put lot of effort into investigation and we found that there are 2 issues:

1. BGP will stop suddenly to propagate routes and windows node can't communicate with k8s nodes at all (weird it happened to win 2 nodes out of 3 - third node no issues at all). Linux nodes no issues at all.
2. Interface metric needs to be adjusted otherwise host network can't communicate over LAN at all.

We're testing flannel right now and we hope we won't see this issue anymore (from tests looks good but not implemented in production yet).

**Comment by song-jiang** [comment]
@MyKEms Sorry for late reply but are you still having this issue?

**Comment by MyKEms** [comment]
I'm closing this issue since we experience same with flannel and we never found root cause. Only reboot of win machine help us by a time when networking is broken.

**Comment by faxioman** [comment]
Even though the ticket is closed, I'm adding a comment for anyone who might land on this issue, like myself. After verifying this issue ourselves with Calico, we transitioned to Flannel, creating a completely custom k8s worker setup script.

We confirmed that, particularly when DSR is enabled, the problem still occurs. It was interesting to discover this script in the Microsoft SDN repository:
```
https://github.com/microsoft/SDN/blob/master/Kubernetes/windows/debug/hnsautomitigator/hnsautomitigator2019.yaml
```
Evidently, Microsoft is also aware of the issue, and I would say that deploying this DaemonSet is currently the only way to have a stable network on Windows workers.
